compiler/rustc_mir/src/transform/separate_const_switch.rs

```diff
@@ -0,0 +1,290 @@
+//! A pass that copies switch-terminated blocks into
+//! another copy for each parent that set the value
+//! being switched over to a constant.
+
+use crate::transform::MirPass;
+use rustc_middle::mir::*;
+use rustc_middle::ty::TyCtxt;
+use smallvec::SmallVec;
+
+//use super::simplify;
+
+pub struct SeparateConstSwitch;
+
+impl<'tcx> MirPass<'tcx> for SeparateConstSwitch {
+    fn run_pass(&self, tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
+        if tcx.sess.mir_opt_level() < 4 {
+            return;
+        }
+
+        separate_const_switch(body);
```

Comment:
You can call `super::simplify::simplify_cfg(body)` directly here if the pass has done anything.

Reply:
I found experimentally that the SimplifyCfg pass is actually unnecessary for the propagation to happen, it's just for some reason limited to functions that are not generic. But it's probably ConstProp's business at that point.

Reply:
This was a stupid comment. Disregard. I added simplify_cfg as suggested.